src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs

```diff
@@ -990,31 +1013,52 @@ private void UpdateBindingContextForSheetContent()
 
         private async Task TranslateBasedOnPosition(double newPosition)
         {
+            // IMPORTANT: Remember to always return when you set position in this method, else the app can go into a recursive forever loop
+
+            if (m_supressNextTranslation)
+                return;
             if (!IsOpen) return;
             if (m_modalityLayout == null) return;
             if (m_sheetView == null) return;
+            
 
-            if (MinPosition < m_autoCloseThreshold || MinPosition > MaxPosition
-            ) //Min position should be bigger than the auto close threshold and max position
+            if (MinPosition > MaxPosition) //Min position should be less than max position
+            {
                 MinPosition = (double)MinPositionProperty.DefaultValue;

```

Comment:
Throw Xaml exception? Magic setting invalid properties might confuse the consumer